src/website/shared/tests/test_auth.py

```diff
@@ -0,0 +1,199 @@
+            metadata=create_metadata_with_maintainer(1, cls.committer.username),
+            parent_evaluation=cls.evaluation,
+        )
+        cls.another_derivation = models.NixDerivation.objects.create(
+            metadata=create_metadata_with_maintainer(1, cls.committer.username),
+            parent_evaluation=cls.evaluation,
+        )
+        cls.derivation_not_related = models.NixDerivation.objects.create(
+            metadata=create_metadata_with_maintainer(2, "not-related"),
+            parent_evaluation=cls.evaluation,
+        )
+
+    def test_superuser_can_add_issue_from_admin_site(self) -> None:
+        self.client.login(username=self.superuser.username, password=self.password)
+        data = {
+            "code": "NIXPKGS-2024-0000",
+            "cve": [self.cve.id],  # type: ignore
+            "description": self.description.id,  # type: ignore
+            "status": "U",
+            "derivations": [self.derivation.id, self.derivation_not_related.id],  # type: ignore
+        }
+
+        response = self.client.post(reverse("admin:shared_nixpkgsissue_add"), data=data)
+        self.assertRedirects(
+            response,
+            status_code=302,
+            expected_url=reverse("admin:shared_nixpkgsissue_changelist"),
+            target_status_code=200,
+        )
+
+        # Make sure that only one issue was created
+        self.assertEqual(models.NixpkgsIssue.objects.count(), 1)
+
+        # And that the changelist view only shows one issue
+        redirect_response = self.client.get(
+            reverse("admin:shared_nixpkgsissue_changelist")
+        )
+        self.assertContains(redirect_response, "1 nixpkgs issue", status_code=200)
+
+    def test_security_can_add_issue_from_admin_site(self) -> None:
+        self.client.login(
+            username=self.security_member.username, password=self.password
+        )
+        data = {
+            "code": "NIXPKGS-2024-0000",
+            "cve": [self.cve.id],  # type: ignore
+            "description": self.description.id,  # type: ignore
+            "status": "U",
+            "derivations": [self.derivation.id, self.derivation_not_related.id],  # type: ignore
+        }
+
+        response = self.client.post(reverse("admin:shared_nixpkgsissue_add"), data=data)
+        self.assertRedirects(
+            response,
+            status_code=302,
+            expected_url=reverse("admin:shared_nixpkgsissue_changelist"),
+            target_status_code=200,
+        )
+
+        # Make sure that only one issue was created
+        self.assertEqual(models.NixpkgsIssue.objects.count(), 1)
+
+        # And that the changelist view only shows one issue
+        redirect_response = self.client.get(
+            reverse("admin:shared_nixpkgsissue_changelist")
+        )
+        self.assertContains(redirect_response, "1 nixpkgs issue", status_code=200)
+
+    def test_committer_cannot_add_non_related_issue_from_admin_site(self) -> None:
```

Comment:
```suggestion
    def test_maintainer_cannot_add_non_related_issue_from_admin_site(self) -> None:
```

This only tests what *maintainers* can do

Reply:
Also there seems to be no test for actual committers.